RecoTracker/MkFitCMS/src/MkStdSeqs.cc

```diff
@@ -0,0 +1,586 @@
+      const float drmax_bl = itrcfg.m_params.c_drmax_bl;
+      const float dzmax_el = itrcfg.m_params.c_dzmax_el;
+      const float drmax_el = itrcfg.m_params.c_drmax_el;
+
+      const float ptmin_hpt = itrcfg.m_params.c_ptthr_hpt;
+
+      const float dzmax2_inv_bh = 1.f / (dzmax_bh * dzmax_bh);
+      const float drmax2_inv_bh = 1.f / (drmax_bh * drmax_bh);
+      const float dzmax2_inv_eh = 1.f / (dzmax_eh * dzmax_eh);
+      const float drmax2_inv_eh = 1.f / (drmax_eh * drmax_eh);
+      const float dzmax2_inv_bl = 1.f / (dzmax_bl * dzmax_bl);
+      const float drmax2_inv_bl = 1.f / (drmax_bl * drmax_bl);
+      const float dzmax2_inv_el = 1.f / (dzmax_el * dzmax_el);
+      const float drmax2_inv_el = 1.f / (drmax_el * drmax_el);
+
+      // Merge hits from overlapping seeds?
+      // For now always true, we require extra hits after seed.
+      const bool merge_hits = true;  // itrcfg.merge_seed_hits_during_cleaning();
+
+      if (seed_ptr == nullptr)
+        return 0;
+      TrackVec &seeds = *seed_ptr;
+
+      const int ns = seeds.size();
+#ifdef DEBUG
+      std::cout << "before seed cleaning " << seeds.size() << std::endl;
+#endif
+      TrackVec cleanSeedTracks;
+      cleanSeedTracks.reserve(ns);
+      std::vector<bool> writetrack(ns, true);
+
+      const float invR1GeV = 1.f / Config::track1GeVradius;
+
+      std::vector<int> nHits(ns);
+      std::vector<int> charge(ns);
+      std::vector<float> oldPhi(ns);
+      std::vector<float> pos2(ns);
+      std::vector<float> eta(ns);
+      std::vector<float> ctheta(ns);
+      std::vector<float> invptq(ns);
+      std::vector<float> pt(ns);
+      std::vector<float> x(ns);
+      std::vector<float> y(ns);
+      std::vector<float> z(ns);
+      std::vector<float> d0(ns);
+      int i1, i2;  //for the sorting
+
+      for (int ts = 0; ts < ns; ts++) {
+        const Track &tk = seeds[ts];
+        nHits[ts] = tk.nFoundHits();
+        charge[ts] = tk.charge();
+        oldPhi[ts] = tk.momPhi();
+        pos2[ts] = std::pow(tk.x(), 2) + std::pow(tk.y(), 2);
+        eta[ts] = tk.momEta();
+        ctheta[ts] = 1.f / std::tan(tk.theta());
+        invptq[ts] = tk.charge() * tk.invpT();
+        pt[ts] = tk.pT();
+        x[ts] = tk.x();
+        y[ts] = tk.y();
+        z[ts] = tk.z();
+        d0[ts] = tk.d0BeamSpot(bspot.x, bspot.y);
+      }
+
+      for (int ts = 0; ts < ns; ts++) {
+        if (not writetrack[ts])
+          continue;  //FIXME: this speed up prevents transitive masking; check build cost!
+
+        const float oldPhi1 = oldPhi[ts];
+        const float pos2_first = pos2[ts];
+        const float Eta1 = eta[ts];
+        const float Pt1 = pt[ts];
+        const float invptq_first = invptq[ts];
+
+        // To study some more details -- need EventOfHits for this
+        int n_ovlp_hits_added = 0;
+        // int  n_ovlp_hits_same_module = 0;
+        // int  n_ovlp_hits_shared = 0;
+        // int  n_ovlp_tracks = 0;
```

Comment:
remove

Reply:
Addressed in Round 1 follow-up PR.
trackreco/cmssw#75 https://github.com/trackreco/cmssw/pull/75/commits/e9834b4a33dab0a34078a4861b825f0e1ebdb64d